# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/tests/mochitest/table/test_struct_ariagrid.html
# Commit: 3c36ad47bcd7
# Full Hash: 3c36ad47bcd7074abc806141d99f58b457b2d8fa
# Author: James Teh <jteh@mozilla.com>
# Date: 2021-03-18 04:15:43
# Description:
#   Bug 1698097: Null check parent content when checking whether a tr should use ARIARowAccessible. r=morgan
#   
#   We assumed parent content wasn't null because we've already checked that there is a parent table accessible, suggesting there's also parent content.
#   However, if the tr is at the top level of a shadow root (but the table is not), parent content will be null, causing a crash.
#   Before the fix for bug 1686123, this was fine because the frame for a shadow root is null and we didn't continue for a null frame.
# ==============================================================================

diff -r c54da9c3d4e1 -r 3c36ad47bcd7 accessible/tests/mochitest/table/test_struct_ariagrid.html
--- a/accessible/tests/mochitest/table/test_struct_ariagrid.html	Wed Mar 17 23:52:11 2021 +0000
+++ b/accessible/tests/mochitest/table/test_struct_ariagrid.html	Thu Mar 18 00:18:29 2021 +0000
@@ -54,6 +54,10 @@
       cellsArray = [ [] ];
       testTableStruct("grid4", cellsArray);
 
+      // ////////////////////////////////////////////////////////////////////////
+      // ARIA table with tr inside a shadow root (bug 1698097).
+      testTableStruct("tableShadow", [ [ kDataCell ] ]);
+
       SimpleTest.finish();
     }
 
@@ -143,5 +147,17 @@
   <!-- Wrong markup ARIA grid -->
   <div role="grid" id="grid3"></div>
   <div role="grid" id="grid4"><div role="row"></div></div>
+
+  <div id="tableShadow" role="table"></div>
+  <script>
+    let host = document.getElementById("tableShadow");
+    let shadow = host.attachShadow({mode: "open"});
+    let tr = document.createElement("tr");
+    shadow.append(tr);
+    tr.setAttribute("role", "row");
+    let td = document.createElement("td");
+    tr.append(td);
+    td.textContent = "test";
+  </script>
 </body>
 </html>
